Assets/MRTK/SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorLoadingBar.cs

```diff
@@ -94,6 +97,25 @@ public async Task CloseAsync()
             gameObject.SetActive(false);
         }
 
+        /// <inheritdoc/>
+        public async Task AwaitTransitionAsync()
+        {
+            while (isActiveAndEnabled)
+            {
+                switch (state)
+                {
+                    case ProgressIndicatorState.Open:
+                    case ProgressIndicatorState.Closed:
+                        return;
+
+                    default:
+                        break;
+                }
+
+                await Task.Yield();
+            }
+        }
+
```

Comment:
nit: this looks like duplicated code. Could this and other be shared in base class?

Reply:
I go back and forth on this. I want to encourage people to implement the interface without relying on a base class because progress indicators can be pretty unique. @keveleigh @julenka Thoughts?

Reply:
I'm going to resolve this for now but I'm open to a future change.